xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -677,22 +679,6 @@ bool CGUIMediaWindow::GetDirectory(const std::string &strDirectory, CFileItemLis
     // took over a second, and not normally cached, so cache it
     if ((XbmcThreads::SystemClockMillis() - time) > 1000  && items.CacheToDiscIfSlow())
       items.Save(GetID());
-
-    // if these items should replace the current listing, then pop it off the top
-    if (items.GetReplaceListing())
-      m_history.RemoveParentPath();
-  }
-
-  // update the view state's reference to the current items
-  m_guiState.reset(CGUIViewState::GetViewState(GetID(), items));
```

Comment:
@xhaggi It breaks views state loading for classes overriding it. What you're looking at is a separate commit btw. This should always be done for all directories, it's not and implementation detail of GetDirectory so I moved it Update.